compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUVectorLowering.cpp

```diff
@@ -142,18 +144,58 @@ struct FoldI1SelectToBroadcast final : OpRewritePattern<arith::SelectOp> {
   }
 };
 
+struct SetMulAddFMF final : OpRewritePattern<arith::AddFOp> {
+  using OpRewritePattern::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(arith::AddFOp addOp,
+                                PatternRewriter &rewriter) const override {
+    auto mulOp = addOp.getLhs().getDefiningOp<arith::MulFOp>();
+    if (!mulOp) {
+      mulOp = addOp.getRhs().getDefiningOp<arith::MulFOp>();
+    }
+
+    if (!mulOp) {
+      return failure();
+    }
+
+    constexpr auto contract = arith::FastMathFlags::contract;
+    if (mulOp.getFastmath() != arith::FastMathFlags::none &&
+        addOp.getFastmath() != arith::FastMathFlags::none) {
+      return failure();
+    }
+
+    rewriter.replaceOpWithNewOp<arith::MulFOp>(
+        mulOp, mulOp.getType(), mulOp.getLhs(), mulOp.getRhs(), contract);
+    rewriter.replaceOpWithNewOp<arith::AddFOp>(
+        addOp, addOp.getType(), addOp.getLhs(), addOp.getRhs(), contract);
```

Comment:
Can we update those in-place using [modifyOpInPlace](https://mlir.llvm.org/doxygen/classmlir_1_1RewriterBase.html#abd8bae753b51386417536a36cf52d3f7)?